src/types/stake.ts

```diff
@@ -6,7 +6,7 @@ import { StakingTypeEnum } from './enums';
  */
 export interface StakingTransactionData {
```

Comment:
This type becomes confusing.

First it's used both for transaction history and for transaction execution. 
- Do "mint" transactions appear in transaction history at all?

For transaction execution it has these properties:
1. amount
  - it's generally supposed to be used when sending tx as value
  - it's defined for stake
  - it's not defined for unstake
  - it should not be defined for mint
2. No need for referrer in the output of transaction methods


I suggest:
1. Use it only for transaction history, closely matching real properties we get from there
2. Either do not use any time when executing transaction - just return untyped object or create 3 different types for each of the transaction. The latter is somewhat worse for the docs, imho.

One of the reasons for this changes is that we don't want the users to have problems with the typescript, guessing did the transaction created by `buildStakeTransaction` return stake or may be mint or unstake type, did it return the amount, and what is that referrer.

Reply:
I agree, the type is confusing, we should probably need different types for every tx, I'll add them.

But for tx history we're using the `VaultTransaction` type, and yes, we do have the mint tx, see `VaultActionType`

Not sure the `referrer` property is not useful - can become at some point, I would keep it.

Reply:
fixed